### PR TITLE
App Configuration Refresh Docs

### PR DESCRIPTION
**Pull request by mrm9084**
Added Refresh Section, fixed visual issues in other sections.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @mrm9084 

How about merge AppConfig related content into 4.0.0-beta.3 branch?

Here are some information current branch strategy:
1.	Reference doc will display 4.0.0-beta.3 by default. i.e. the following links are the same page:
1.1.	https://microsoft.github.io/spring-cloud-azure
1.2.	https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html
1.3.	https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.3/4.0.0-beta.3/reference/html/index.html
2.	For both 4.0.0-beta.3 and 4.0.0-beta.4 branch, when the contents updated, GitHub actions will update related reference docs. Here is the link to related docs page:
2.1.	4.0.0-beta.3: https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.3/4.0.0-beta.3/reference/html/index.html
2.2.	4.0.0-beta.4: https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html
3.	For 4.0.0-beta.3, the “_configprops.adoc” file will updated according to the 4.0.0-beta.3 code. But for 4.0.0-beta.4, the “_configprops.adoc” is still using the property of 4.0.0-beta.3. Refer to this commit to get more information.
4.	Both “4.0.0-beta.3” and “4.0.0-beta.4” are protected branches. If you want to update the reference doc, please create PR instead of push directly.
5.	The GitHub default branch changed from “main” to “4.0.0-beta.3”, which is the latest released version.
6.	If you want to update content in both 4.0.0-beta.3 and 4.0.0-beta.4, please create PR target to 4.0.0-beta.3, then merge 4.0.0-beta.3 into 4.0.0-beta.4. Here are the example PRs:
6.1.	[Create PR to 4.0.0-beta.3](https://github.com/microsoft/spring-cloud-azure/pull/814).
6.2.	[Merge 4.0.0-beta3 into 4.0.0-beta.4](https://github.com/microsoft/spring-cloud-azure/pull/814).

**Comment by mrm9084** [comment]
@chenrujun We talked about this a few weeks ago. We decided not to merge into the 4.0.0 branch until app config had a 4.0.0 release.

**Review by chenrujun — APPROVED** [review]
LGTM.

**Comment by chenrujun** [comment]
@mrm9084 . Got it.
I approved this PR, you can merge it now.

File: docs/src/main/asciidoc/refresh.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+----
+https://www.myaplication.com/actuator/appconfiguration-refresh?myTokenName=mySecret
+----
+
+In order to "Confirm Selection" will send a setup notification to the given URI and it expects a response. If no response is returned it will fail. The `azure-spring-cloud-appconfiguration-web` library if setup for endpoints will return the correct response if the Azure App Configuration store is configured for the application. This confirmation can be sent in other ways see https://docs.microsoft.com/azure/event-grid/webhook-event-delivery[Event Grid Web Hook Delivery] for more information on Web Hook validation.
+
+NOTE: This validation only happens on the creation/modification of the endpoint.
+
+It is highly recommended that filters are sutp as otherwise a refresh will be triggered after every key creation and modification.
```
Comment (AlexandraKemperMS) [inline docs/src/main/asciidoc/refresh.adoc:70]: Word: set up

File: docs/src/main/asciidoc/refresh.adoc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+[,console,indent=0]
+----
+INFO 17496 --- [TaskScheduler-1] o.s.c.e.event.RefreshEventListener       : Refresh keys changed: [config.message]
+----
+
+The application will refresh all `@Bean`++'++s in the refresh scope. By default, `@ConfiugrationProperties` annotated beans will be included in this scope.
+
+=== Pull Based Refresh
+
+The library supports the ability to periodically check on a refresh interval for change made to the monitoring triggers. By default the refresh-interval is set to 30 seconds. Once the refresh interval has passed all triggers will be checked in the given store for changes. Any update to the value will cause a refresh to trigger. Because the libraries integrate with the Spring refresh system any refresh will reload configurations from all stores.
```
[the remark this answers is not in the document]
Comment (mrm9084) [reply docs/src/main/asciidoc/refresh.adoc:27]: For the monitoring triggers I updates the section that they are defined in with this info.

File: docs/src/main/asciidoc/refresh.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Also, because the library uses Spring's configuration system, triggering a refresh will also cause a refresh of all of your configurations, not just reloading the ones from your Azure App Configuration store.
+
+=== Push Based Refresh
+
+The `azure-spring-cloud-appconfiguration-config-web` library can be setup to receive push notifications from your Azure App Configuration store to refresh your configuration values. This is done via an Azure Event Grid Web Hook, which can be configured to send notifications of changes to specified keys. By adding the Spring Actuator library as a dependency you can expose App Configuration's refresh endpoint(s). There are two different endpoints, `appconfiguration-refresh` and `appconfiguration-refresh-bus` these endpoints work sort of similarly to there counterparts `refresh` and `refresh-bus`, where the app configuration endpoints expire the refresh interval instead of forcing a refresh upon receiving. The `refresh` and `refresh-bus` can still be used, but can't be directly connected to Azure Event Grid with a Web Hook as they require a response in setup.
```
Comment (AlexandraKemperMS) [inline docs/src/main/asciidoc/refresh.adoc:48]: there counterparts -> their counterparts

File: docs/src/main/asciidoc/refresh.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+`appconfiguration-refresh` expires the refresh interval, so the remaining refresh interval isn't waited on before the next refresh check. `appconfiguration-refresh-bus` will send a notification to a connected messaging service, such as Azure Service Bus to notify all instances of an application to refresh. In both cases it doesn't completely expire the refresh interval, but almost does by a small jitter amount. This makes sure not every instance of your application doesn't try to refresh at the same time.
```
Comment (AlexandraKemperMS) [inline docs/src/main/asciidoc/refresh.adoc:50]: Last sentence is not clear - I think you mean "tries" instead of "doesn't try"?

Comment (mrm9084) [reply docs/src/main/asciidoc/refresh.adoc:50]: I think it is the double negative that is the confusing part. I removed the not.

**Pull request merged**